api/src/gameState/gameState.controller.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  Param,
+  ParseIntPipe,
+  Post,
+  Put,
+  NotFoundException,
+} from '@nestjs/common';
+import { AppGateway } from '../app.gateway';
+import { GameState, Player, Node } from '../types';
+import { GameStateService } from './gameState.service';
+import { INITIAL_BOARD } from '../constants';
+import { Game } from '@prisma/client';
+
+@Controller('games')
+export class GameStateController {
+  constructor(
+    private readonly gameStateService: GameStateService,
+    private gatewayService: AppGateway,
+  ) {}
+
+  @Post('')
+  async createGame(): Promise<GameState> {
+    const res = await this.gameStateService.createGame();
+    return this.gameStateService.deserializer(res);
+  }
+
+  @Get(':id')
+  async getGameState(
+    @Param('id', ParseIntPipe) id: number,
+  ): Promise<GameState> {
+    let res: Game;
+    try {
+      res = await this.gameStateService.getGame({ id });
+    } catch (error) {
+      throw new NotFoundException("That game id doesn't exists");
+    }
+    return this.gameStateService.deserializer(res);
+  }
+
+  @Put(':id/join')
+  async joinGame(@Param('id', ParseIntPipe) id: number): Promise<GameState> {
+    let res: Game;
+    try {
+      res = await this.gameStateService.getGame({ id });
+    } catch (e) {
+      throw new NotFoundException("That game id doesn't exists");
+    }
+    const gameState: GameState = this.gameStateService.deserializer(res);
+    return gameState;
+  }
+
+  @Post(':id/restart')
+  async restartGame(@Param('id', ParseIntPipe) id: number): Promise<GameState> {
+    let gameState: GameState;
+    try {
+      gameState = await this.gameStateService.restartGame(id);
+    } catch (e) {
+      throw new NotFoundException("That game doesn't exists");
+    }
+
+    this.gatewayService.emitGameState(gameState);
+    return gameState;
+  }
+
+  @Put(':id/marble-clicked')
+  async setMarbleClicked(
+    @Param('id', ParseIntPipe) id: number,
+    @Body('marbleClicked') marbleClicked: Node,
+  ) {
+    let res: Game;
+    try {
+      res = await this.gameStateService.updateGame({
+        where: { id },
+        data: {
+          marbleClicked: JSON.stringify(marbleClicked),
+        },
+      });
+    } catch (error) {
+      throw new Error('unable to update the game');
+    }
+    const gameState = this.gameStateService.deserializer(res);
+    this.gatewayService.emitGameState(gameState);
+    return gameState;
+  }
+
+  @Post(':id/authorized-move')
```

Comment:
Post? Do you really modify the state of a game?

Reply:
No, but at this time, we need to pass some Data, as the playerNumber, the direction and the game id. 
For now, making the request as a GET will transform a call into something like this : `games/ID/authorized-move/PLAYERID/DIRECTION`.
I know that POST will induce modifying something on the server side, but i thought i would be clearer like that.
In fact in the other PR `API-AUTH` we can retrieve the player id by the token sent. At this moment, we'll only have to pass 2 params in the GET.